community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/ast/convert/plannerQuery/ClauseConverters.scala

```diff
@@ -365,14 +362,26 @@ object ClauseConverters {
     }
   }
 
-    def addRemoveToLogicalPlanInput(acc: PlannerQueryBuilder, clause: Remove): PlannerQueryBuilder = {
-      clause.items.foldLeft(acc) {
-        // REMOVE n:Foo
-        case (builder, RemoveLabelItem(identifier, labelNames)) =>
-          builder.amendUpdateGraph(ug => ug.addMutatingPatterns(RemoveLabelPattern(IdName.fromVariable(identifier), labelNames)))
-
-        case (builder, other) =>
-          throw new CantHandleQueryException(s"REMOVE $other not supported in cost planner yet")
-      }
+  private def addRemoveToLogicalPlanInput(acc: PlannerQueryBuilder, clause: Remove): PlannerQueryBuilder = {
+    clause.items.foldLeft(acc) {
+      // REMOVE n:Foo
+      case (builder, RemoveLabelItem(variable, labelNames)) =>
+        builder.amendUpdateGraph(ug => ug.addMutatingPatterns(RemoveLabelPattern(IdName.fromVariable(variable), labelNames)))
+
+      // REMOVE n.prop
+      case (builder, RemovePropertyItem(Property(variable: Variable, propertyKey))) if acc.semanticTable.isNode(variable) =>
+        builder.amendUpdateGraph(ug => ug.addMutatingPatterns(
+          SetNodePropertyPattern(IdName.fromVariable(variable), propertyKey, Null()(propertyKey.position))
+        ))
+
+      // REMOVE rel.prop
+      case (builder, RemovePropertyItem(Property(variable: Variable, propertyKey))) if acc.semanticTable.isRelationship(variable) =>
+        builder.amendUpdateGraph(ug => ug.addMutatingPatterns(
+          SetRelationshipPropertyPattern(IdName.fromVariable(variable), propertyKey, Null()(propertyKey.position))
+        ))
+
+      case (builder, other) =>
+        throw new CantHandleQueryException(s"REMOVE $other not supported in cost planner yet")
```

Comment:
This should probably be a different exception; we only support `REMOVE` for labels and properties; if something else comes in here there's some real shenanigans going on.